r2/r2/lib/providers/search/__init__.py

```diff
@@ -20,14 +20,28 @@
 # Inc. All Rights Reserved.
 ###############################################################################
 
-import r2.lib.cloudsearch as cloudsearch
 
+class SearchProvider(object):
+    """Provider for search.
+    """
 
-InvalidQuery = (cloudsearch.InvalidQuery,)
-SearchException = (cloudsearch.CloudSearchHTTPError,)
+    def InvalidQuery(self):
+        return NotImplementedError
 
-SearchQuery = cloudsearch.LinkSearchQuery
-SubredditSearchQuery = cloudsearch.SubredditSearchQuery
+    def SearchException(self):
+        return NotImplementedError
 
-sorts = cloudsearch.LinkSearchQuery.sorts_menu_mapping
-sr_sorts = cloudsearch.SubredditSearchQuery.sorts_menu_mapping
+    def Query(self):
+        return NotImplementedError
+
+    def SubredditSearchQuery(self):
+        return NotImplementedError
+
+    def sorts(self):
+        return NotImplementedError
+
+    def run_changed(self):
+        return NotImplementedError
+
+    def get_related_query(self):
+        return NotImplementedError
```

Comment:
All these functions should `raise NotImplementedError` rather than returning it.

(`return NotImplemented` is a pattern more used for implementing comparators, I believe)